src/lib/temple/front/address-book.ts

```diff
@@ -1,44 +1,70 @@
-import { useCallback } from 'react';
+import { useCallback, useMemo } from 'react';
 
 import { getMessage } from 'lib/i18n';
 import { TempleContact } from 'lib/temple/types';
+import { useSettings } from 'temple/front/ready';
 
 import { useTempleClient } from './client';
-import { useFilteredContacts } from './use-filtered-contacts.hook';
 
 export function useContactsActions() {
   const { updateSettings } = useTempleClient();
-  const { contacts, allContacts } = useFilteredContacts();
+  const { contacts } = useSettings();
+
+  const contactsWithFallback = useMemo(() => contacts ?? [], [contacts]);
 
   const addContact = useCallback(
-    async (cToAdd: TempleContact) => {
-      if (allContacts.some(c => c.address === cToAdd.address)) {
-        throw new Error(getMessage('contactWithTheSameAddressAlreadyExists'));
+    async (contactToAdd: TempleContact) => {
+      checkIfContactAlreadyExists(contactsWithFallback, contactToAdd.address);
+
+      await updateSettings({
+        contacts: [contactToAdd, ...contactsWithFallback]
+      });
+    },
+    [contactsWithFallback, updateSettings]
+  );
+
+  const editContact = useCallback(
+    async (address: string, editedData: Pick<TempleContact, 'name' | 'address'>) => {
+      if (address !== editedData.address) {
+        checkIfContactAlreadyExists(contactsWithFallback, editedData.address);
       }
 
+      const newContacts = [...contactsWithFallback];
+
+      const index = contactsWithFallback.findIndex(c => c.address === address);
+
+      if (index === -1) {
+        throw new Error('Failed to find a contact to edit');
+      }
+
+      const currentContact = contactsWithFallback[index];
+
+      newContacts[index] = { ...currentContact, ...editedData };
+
       await updateSettings({
-        contacts: [cToAdd, ...contacts]
+        contacts: newContacts
       });
     },
-    [contacts, allContacts, updateSettings]
+    [contactsWithFallback, updateSettings]
   );
 
   const removeContact = useCallback(
-    (address: string) =>
-      updateSettings({
-        contacts: contacts.filter(c => c.address !== address)
+    async (address: string) =>
+      await updateSettings({
+        contacts: contactsWithFallback.filter(c => c.address !== address)
```

Comment:
```suggestion
    (address: string) =>
      updateSettings({
        contacts: contactsWithFallback.filter(c => c.address !== address)
```

Reply:
why? The list filtering happening in real time after modal close, if not awaited, it looks a bit weird

Reply:
You have to await in a function that calls it anyway, so everything should be fine if you remove these words